### PR TITLE
Docs/doc 95/doc 116/technical term data docs

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Adds technical term page for Data Docs.

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation

Thank you for submitting!

### CODE REVIEW

File: docs/terms/data_docs.md [hunk 1]
```diff
@@ -1,14 +1,100 @@
...
+context.build_data_docs()
+```
+
+### Configuration
+
+Data Docs sites are configured under the `data_docs_sites` key in your deployment's `great_expectations.yml` file. Users can specify:
+
+- which <TechnicalTag relative="../" tag="datasource" text="Datasources" /> to document (by default, all)
+- whether to include Expectations, validations and profiling results sections
+- where the Expectations and validations should be read from (filesystem, S3, or GCS)
```
Comment (cdkini) [inline docs/terms/data_docs.md]: Could we also Azure if we're including the other cloud providers? Same below.

File: docs/terms/data_docs.md [hunk 2]
```diff
@@ -1,14 +1,100 @@
...
+
+### How to access
+
+Data Docs are rendered as HTML files.  As such, you can open them with any browser.
+
+### How to create
+
+If your Data Docs have not yet been rendered, you can create them from your Data Context.
+
+From the root folder of your Data Context, you can build your Data Docs with the CLI command:
```
Comment (cdkini) [inline docs/terms/data_docs.md]: Might we want to say "the root folder of your project"? I could see users thinking the root folder of `great_expectations/`, which would not work as intended.

File: docs/terms/data_docs.md [hunk 3]
```diff
@@ -1,14 +1,100 @@
...
+1. Maintain a "shared source of truth" for a team working on a data project. Such documentation renders all the
+   artifacts committed in the source control system (Expectation Suites and profiling results) and a continuously
+   updating data quality report, built from a chronological list of validations by run id.
+1. Share a spec of a dataset with a client or a partner. This is similar to API documentation in software development.
+   This documentation would include profiling results of the dataset to give the reader a quick way to grasp what the
+   data looks like, and one or more Expectation Suites that encode what is expected from the data to be considered
+   valid.
+
+To support these (and possibly other) use cases Great Expectations has a concept of "data documentation site". Multiple sites can be configured inside a project, each suitable for a particular data documentation use case.
+
```
Comment (cdkini) [inline docs/terms/data_docs.md]: Should we have a comma after use-case?

**Review by cdkini — APPROVED** [review]
LGTM but made a comment or two that might be of value!

File: docs/terms/data_docs.md (hunk 1 above)
Comment (Rachel-Reverie) [reply docs/terms/data_docs.md]: Added to both locations.

File: docs/terms/data_docs.md (hunk 2 above)
Comment (Rachel-Reverie) [reply docs/terms/data_docs.md]: I updated it to say the root folder of their project, and specified that it was where their Data Context would have been initialized from.

File: docs/terms/data_docs.md (hunk 3 above)
Comment (Rachel-Reverie) [reply docs/terms/data_docs.md]: A comma could be included there, but isn't necessary.

Comment (cdkini) [reply docs/terms/data_docs.md]: oh English why are you so tricky

**Pull request merged**